### PR TITLE
net: openthread: allow to configure `OPENTHREAD_MESSAGE_BUFFER_SIZE`

### PR DESCRIPTION
**Pull request by edmont**
Add `OPENTHREAD_CONFIG_MESSAGE_BUFFER_SIZE` to Kconfig.

Also set the number of children to minumum possible for MTD builds
in order to save some resources (~512B of RAM).

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

### CODE REVIEW

File: subsys/net/l2/openthread/Kconfig [hunk 1]
```diff
@@ -265,6 +265,12 @@ config OPENTHREAD_NUM_MESSAGE_BUFFERS
 	help
 	  "The number of message buffers in the buffer pool."
 
+config OPENTHREAD_MESSAGE_BUFFER_SIZE
+	int "The size of a message buffer in bytes"
+	default 128
```
Comment (pdgendt) [inline subsys/net/l2/openthread/Kconfig:270]: Existing applications that pull in this change, might not work as expected? I like the option, just not sure if we should reduce it as default.

**Review by pdgendt — APPROVED** [review]

File: subsys/net/l2/openthread/Kconfig [hunk 2]
```diff
@@ -265,6 +265,12 @@ config OPENTHREAD_NUM_MESSAGE_BUFFERS
 	help
 	  "The number of message buffers in the buffer pool."
 
+config OPENTHREAD_MESSAGE_BUFFER_SIZE
+	int "The size of a message buffer in bytes"
```
[the remark this answers is not in the document]
Comment (edmont) [reply subsys/net/l2/openthread/Kconfig:269]: Just being consistent with the above symbol. Maybe this can be adapted in a separated PR?

File: subsys/net/l2/openthread/Kconfig (hunk 1 above)
Comment (edmont) [reply subsys/net/l2/openthread/Kconfig:270]: Default value in OpenThread is 128 bytes for 32bit architectures and 256 bytes for 64bit architectures. Are you aware of any Kconfig option to detect the system's CPU word length?

In any case static asserts in `message.hpp` and `coap_message.hpp` will detect any issues with the buffer size at compile time.

File: modules/openthread/Kconfig.thread [hunk 3]
```diff
@@ -61,9 +61,9 @@ config OPENTHREAD_POLL_PERIOD
 
 config OPENTHREAD_MAX_CHILDREN
 	int "The maximum number of children"
-	range 10 511
+	range 1 511
+	default 1 if OPENTHREAD_MTD
```
Comment (rlubos) [inline modules/openthread/Kconfig.thread:65]: Can you elaborate a bit on why is this needed? Does OT still allocate some children-specific resources even for MTD devices?

**Review by rlubos — APPROVED** [review]

File: modules/openthread/Kconfig.thread (hunk 3 above)
Comment (edmont) [reply modules/openthread/Kconfig.thread:65]: Yes, for instance the `Metadata` struct of message `Buffer` always contains the `ChildMask mChildMask` which is size dependent on the number of children:

`typedef BitVector<OPENTHREAD_CONFIG_MLE_MAX_CHILDREN> ChildMask;`

There are other cases as well.
